### ISSUE TITLE
Need Pytorch version of Bert using Adam optimizer for benchmarking

### ISSUE THREAD

**Issue by David-Levinthal** [NVIDIA/DeepLearningExamples#241]
Currently this is not an option as it is in the Tensorflow tree. It appears that at least 2 changes to run_pretraining.py would be needed:
from optimization import BertLAMB
to become
from optimization import BertAdam
and at line 307
   optimizer = BertLAMB(optimizer_grouped_parameters,
                         lr=args.learning_rate,
                         warmup=args.warmup_proportion,
                         t_total=args.max_steps)
to become
   optimizer = BertAdam(optimizer_grouped_parameters,
                         lr=args.learning_rate,
                         warmup=args.warmup_proportion,
                         t_total=args.max_steps)
and then quite a few changes to scripts/run_pretraining.sh

what else would be needed?

**Comment by sharathts** [comment]
BertAdam would break with AMP.

You can use/follow the pretraining example with apex FusedAdam published in this version: https://github.com/NVIDIA/DeepLearningExamples/blob/a644350589f9abc91b203f73e686a50f5d6f3e96/PyTorch/LanguageModeling/BERT

**Comment by David-Levinthal** [comment]
what about a pytorch version using adam with manual_fp16 optimizations?
this speeds up the Tensorflow version considerably

**Comment by sharathts** [comment]
FusedAdam works with AMP optimization level O2 and is faster or at least as fast as our TF implementation with manual_fp16 optimizations.
